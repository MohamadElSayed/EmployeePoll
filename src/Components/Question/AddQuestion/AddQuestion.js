import Menu from "../../Menu";
import { connect } from "react-redux";
import { useState } from "react";
import { handleSaveQuestion } from "../../../actions/questions";
import { useNavigate } from "react-router-dom";

const AddQuestion = ({ authedUser, dispatch }) => {
  const [optionOneText, setOptionOneText] = useState("");
  const [optionTwoText, setOptionTwoText] = useState("");

  const navigate = useNavigate();

  const handleSubmit = (event) => {
    event.preventDefault();
    dispatch(
      handleSaveQuestion({
        author: authedUser.id,
        optionOneText: optionOneText,
        optionTwoText: optionTwoText,
      })
    );
    navigate("/", { replace: true });
  };

  return (
    <div className="container text-center">
      <Menu />
      <form
        className="justify-content-center"
        onSubmit={(e) => handleSubmit(e)}
      >
        <div className="row">
          <h3
            className="col"
            style={{ textAlign: "center", paddingTop: "50px" }}
          >
            Would You Rather
          </h3>
        </div>
        <div className="row">
          <h5
            className="col"
            style={{ textAlign: "center", paddingTop: "20px" }}
          >
            Create Your own Poll
          </h5>
        </div>

        <div className="row">
          <div
            className="col"
            style={{ textAlign: "center", paddingTop: "10px" }}
          >
            First Option
          </div>
        </div>

        <div className="row justify-content-center">
          <div className="col" style={{ paddingTop: "10px" }}>
            <input
              type="text"
              name="FirstOption"
              className="form-control"
              style={{
                width: "70%",
                display: "block",
                marginRight: "auto",
                marginLeft: "auto",
              }}
              value={optionOneText}
              placeholder="Option 1 Text"
              onChange={(e) => setOptionOneText(e.target.value)}
            />
          </div>
        </div>

        <div className="row">
          <div
            className="col"
            style={{ textAlign: "center", paddingTop: "10px" }}
          >
            Second Option
          </div>
        </div>

        <div className="row justify-content-center">
          <div className="col" style={{ paddingTop: "10px" }}>
            <input
              type="text"
              name="SecondOption"
              className="form-control"
              style={{
                width: "70%",
                display: "block",
                marginRight: "auto",
                marginLeft: "auto",
              }}
              value={optionTwoText}
              onChange={(e) => setOptionTwoText(e.target.value)}
              placeholder="Option 2 Text"
            />
          </div>
        </div>

        <br />
        <br />
        <div className="row justify-content-center">
          <div className="col" style={{ paddingTop: "10px" }}>
            <button
              type="submit"
              name="submitNewPoll"
              className="btn btn-primary "
              style={{
                width: "20%",
                display: "block",
                marginRight: "auto",
                marginLeft: "auto",
              }}
              disabled={
                optionOneText.trim() === "" || optionTwoText.trim() === ""
              }
            >
              Submit{" "}
            </button>
          </div>
        </div>
      </form>
    </div>
  );
};

export default connect((state) => ({
  authedUser: state.authedUser,
}))(AddQuestion);
