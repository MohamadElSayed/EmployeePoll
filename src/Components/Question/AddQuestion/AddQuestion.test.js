import * as React from "react";
import AddQuestion from "./AddQuestion";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../../../reducers";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

const store = createStore(reducer);

describe("Add Question Snapshot Tests", () => {
  let component;
  beforeEach(() => {
    component = render(
      <Provider store={store}>
        <BrowserRouter>
          <AddQuestion />
        </BrowserRouter>
      </Provider>
    );
  });

  it("matching the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("check that Option1 and Option2 are present", () => {
    let firstOption = component.getByText("First Option");
    expect(firstOption).toBeInTheDocument();

    let secondOption = component.getByText("Second Option");
    expect(secondOption).toBeInTheDocument();

    let submitButton = component.getByText("Submit");
    expect(submitButton).toBeInTheDocument();
  });
});
