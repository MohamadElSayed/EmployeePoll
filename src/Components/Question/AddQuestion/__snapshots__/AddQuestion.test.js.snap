// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Question Snapshot Tests matching the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container text-center"
      >
        <div
          class="container"
        >
          <br />
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Home
                </div>
                <div
                  class="col"
                >
                  <a
                    href="/leaderboard"
                  >
                    Leaderboard
                  </a>
                </div>
                <div
                  class="col"
                >
                  <a
                    href="/add"
                  >
                    New
                  </a>
                </div>
              </div>
            </div>
            <div
              class="col"
            />
            <div
              class="col-md-auto text-end"
            >
              <img
                alt=""
                src=""
                style="width: 20px; height: 20px;"
              />
            </div>
            <div
              class="col col-lg-2 text-end"
            >
              <a
                href="/"
              >
                Logout
              </a>
            </div>
          </div>
          <hr />
        </div>
        <form
          class="justify-content-center"
        >
          <div
            class="row"
          >
            <h3
              class="col"
              style="text-align: center; padding-top: 50px;"
            >
              Would You Rather
            </h3>
          </div>
          <div
            class="row"
          >
            <h5
              class="col"
              style="text-align: center; padding-top: 20px;"
            >
              Create Your own Poll
            </h5>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
              style="text-align: center; padding-top: 10px;"
            >
              First Option
            </div>
          </div>
          <div
            class="row justify-content-center"
          >
            <div
              class="col"
              style="padding-top: 10px;"
            >
              <input
                class="form-control"
                name="FirstOption"
                style="width: 70%; display: block; margin-right: auto; margin-left: auto;"
                type="text"
              />
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
              style="text-align: center; padding-top: 10px;"
            >
              Second Option
            </div>
          </div>
          <div
            class="row justify-content-center"
          >
            <div
              class="col"
              style="padding-top: 10px;"
            >
              <input
                class="form-control"
                name="SecondOption"
                style="width: 70%; display: block; margin-right: auto; margin-left: auto;"
                type="text"
              />
            </div>
          </div>
          <br />
          <br />
          <div
            class="row justify-content-center"
          >
            <div
              class="col"
              style="padding-top: 10px;"
            >
              <button
                class="btn btn-primary "
                name="submitNewPoll"
                style="width: 20%; display: block; margin-right: auto; margin-left: auto;"
                type="submit"
              >
                Submit
                 
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container text-center"
    >
      <div
        class="container"
      >
        <br />
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                Home
              </div>
              <div
                class="col"
              >
                <a
                  href="/leaderboard"
                >
                  Leaderboard
                </a>
              </div>
              <div
                class="col"
              >
                <a
                  href="/add"
                >
                  New
                </a>
              </div>
            </div>
          </div>
          <div
            class="col"
          />
          <div
            class="col-md-auto text-end"
          >
            <img
              alt=""
              src=""
              style="width: 20px; height: 20px;"
            />
          </div>
          <div
            class="col col-lg-2 text-end"
          >
            <a
              href="/"
            >
              Logout
            </a>
          </div>
        </div>
        <hr />
      </div>
      <form
        class="justify-content-center"
      >
        <div
          class="row"
        >
          <h3
            class="col"
            style="text-align: center; padding-top: 50px;"
          >
            Would You Rather
          </h3>
        </div>
        <div
          class="row"
        >
          <h5
            class="col"
            style="text-align: center; padding-top: 20px;"
          >
            Create Your own Poll
          </h5>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
            style="text-align: center; padding-top: 10px;"
          >
            First Option
          </div>
        </div>
        <div
          class="row justify-content-center"
        >
          <div
            class="col"
            style="padding-top: 10px;"
          >
            <input
              class="form-control"
              name="FirstOption"
              style="width: 70%; display: block; margin-right: auto; margin-left: auto;"
              type="text"
            />
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
            style="text-align: center; padding-top: 10px;"
          >
            Second Option
          </div>
        </div>
        <div
          class="row justify-content-center"
        >
          <div
            class="col"
            style="padding-top: 10px;"
          >
            <input
              class="form-control"
              name="SecondOption"
              style="width: 70%; display: block; margin-right: auto; margin-left: auto;"
              type="text"
            />
          </div>
        </div>
        <br />
        <br />
        <div
          class="row justify-content-center"
        >
          <div
            class="col"
            style="padding-top: 10px;"
          >
            <button
              class="btn btn-primary "
              name="submitNewPoll"
              style="width: 20%; display: block; margin-right: auto; margin-left: auto;"
              type="submit"
            >
              Submit
               
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
